include/glow/Importer/ONNX.h

```diff
@@ -41,22 +41,41 @@ class ONNXModelLoader
   /// Load the network initializers from the GraphProto.
   void loadInitializers(onnx::GraphProto &net);
 
+  /// \returns true if operator \p op can be loaded.
   /// Load the operator \p op into the network. This creates one or more nodes
   /// in the network.
-  void loadOperator(const onnx::NodeProto &op);
+  bool loadOperator(const onnx::NodeProto &op);
 
-  /// Reads a network (weights or structure) from the serialized protocol buffer
-  /// file.
-  bool loadProtoFile(onnx::GraphProto &net, const std::string &filename);
+  /// \returns true if serialized protobuf can be read from the \p filename.
+  /// Loads GraphProto \p net containing network (weights or structure)
+  /// from the serialized protobuf file.
+  bool loadProto(onnx::GraphProto &net, const std::string &filename);
 
+  /// \returns true if serialized protobuf can be constructed from the in-memory
+  /// serialized protobuf.
+  /// Loads GraphProto \p net from the in-memory serialized protobuf \p onnxModel
+  /// with the model size \p onnxModelSize.
+  bool loadProto(onnx::GraphProto &net, void const *onnxModel, size_t onnxModelSize);
+
+  /// \returns true if GraphProto \p net can be loaded from the stream \p istream.
+  bool loadProto(onnx::GraphProto &net, google::protobuf::io::ZeroCopyInputStream& istream);
+
+  /// Creates the ONNX model loader with the network which will be stored
+  /// in the \p F. 
+  ONNXModelLoader(Function &F);
 public:
   /// Loads the ONNX model that's represented by a model description file,
   /// serialized in \p modelDescFilename and populates the network into \p F.
   /// The tensors in \p tensors are stored with the names in the list of names
-  /// \p names and used as inputs to the network.
+  /// \p tensorNames and used as inputs to the network.
   ONNXModelLoader(const std::string &modelDescFilename,
-                  llvm::ArrayRef<const char *> names,
+                  llvm::ArrayRef<const char *> tensorNames,
                   llvm::ArrayRef<Tensor *> tensors, Function &F);
+
+  /// \returns unique pointer to ONNXModelLoader if \p onnxModel can be parsed, e.g., the model is the
+  /// valid ONNX model and Glow supports all of the operators in the network.
+  /// \returns nullptr otherwise.
+  static std::unique_ptr<ONNXModelLoader> parse(void const *onnxModel, size_t onnxModelSize, Function &F);
 };
```

Comment:
Is there any test to use this function so far?

Reply:
Yeah, not now. I wish we'd have general Loader test coverage.
Once i'll make more progress on ONNXIFI we'll have integration test calling in here.